File: apps/api-go/internal/domain/model/inspiration.go
package model

import "time"

// JobStatus tracks the AI inspiration pipeline lifecycle.
type JobStatus string

const (
	JobPending    JobStatus = "pending"
	JobSearching  JobStatus = "searching"
	JobGenerating JobStatus = "generating"
	JobReady      JobStatus = "ready"
	JobFailed     JobStatus = "failed"
)

type ReferenceAd struct {
	Body        string
	Title       string
	SnapshotURL string
	Impressions int
}

type GeneratedImage struct {
	URL       string
	RequestID string
}

type Copy struct {
	Headline string
	Body     string
	CTA      string
}

type GeneratedAssets struct {
	Images []GeneratedImage
	Copies []Copy
}

type InspirationJob struct {
	ID                   string
	ClientID             string
	Keywords             []string
	Country              string
	Platforms            []string
	Status               JobStatus
	ReferenceAds         []ReferenceAd
	Assets               GeneratedAssets
	HiggsfieldRequestIDs []string
	Error                string
	CreatedAt            time.Time
}

File: apps/api-nest/src/domain/models/inspiration.ts
export type JobStatus = 'pending' | 'searching' | 'generating' | 'ready' | 'failed';

export interface ReferenceAd {
  body: string;
  title: string;
  snapshotUrl: string;
  impressions: number;
}

export interface GeneratedImage {
  url: string;
  requestId: string;
}

export interface Copy {
  headline: string;
  body: string;
  cta: string;
}

export interface GeneratedAssets {
  images: GeneratedImage[];
  copies: Copy[];
}

export interface InspirationJob {
  id: string;
  clientId: string;
  keywords: string[];
  country: string;
  platforms: string[];
  status: JobStatus;
  referenceAds: ReferenceAd[];
  assets: GeneratedAssets;
  higgsfieldRequestIds: string[];
  error?: string | null;
  createdAt: Date;
}
